lib/session/call-session.js

```diff
@@ -2601,6 +2604,26 @@ Duration=${duration} `
     }
   }
 
+  async _enableInbandDtmf(ep) {
+    if (ep.inbandDtmfEnabled) return;
+    // only enable inband dtmf detection if voip carrier dtmf_type === tones
+    const voip_carrier_sid = this.req.has('X-Voip-Carrier-Sid') ? this.req.get('X-Voip-Carrier-Sid') :
+      this.req.has('X-Requested-Carrier-Sid') ? this.req.get('X-Requested-Carrier-Sid') : null;
+    if (voip_carrier_sid) {
+      const {lookupVoipCarrierBySid} = dbUtils(this.logger, this.srf);
+      const voipCarrier = await lookupVoipCarrierBySid(voip_carrier_sid);
+      if (voipCarrier?.dtmf_type === 'tones') {
+        // https://developer.signalwire.com/freeswitch/FreeSWITCH-Explained/Modules/mod-dptools/6587132/#0-about
+        try {
+          ep.execute('start_dtmf');
+          ep.inbandDtmfEnabled = true;
```

Comment:
why do we need to set this property on ep?

Reply:
Because we are runinging this check on every new task as there is some usecase that the media endpoint might be already exist, might be not created yet, or being replaced.

so that If this flag is enabled for that endpoint, we don't need to set the start_dtmf anymore.